ghapi/actions.py

```diff
@@ -159,3 +160,21 @@ def set_git_user(api=None):
         email = 'github-actions[bot]@users.noreply.github.com'
     run(f'git config --global user.email "{email}"')
     run(f'git config --global user.name  "{user}"')
+
+# %% ../01_actions.ipynb 52
+class PagesNotEnabled(Exception): pass
+class PagesConfigError(Exception): pass
+
+def enable_pages_actions(api):
+    from urllib.error import HTTPError
+    msg="Please enable GitHub Pages to publish from the root of the `gh-pages` branch per these instructions - https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-from-a-branch"
+    try: res = api.enable_pages(branch='gh-pages')
+    except TokenNotAuthorized as e: raise e
```

Comment:
With `TokenNotAuthorized` in GitHub Actions, I would print something letting the user know about the error, but also telling them how to skip this enable pages step, incase they are in a situation where they cannot access a token with enough permissions to do things.